apps/src/code-studio/calendarRedux.ts

```diff
@@ -0,0 +1,37 @@
+import {PayloadAction, createSlice} from '@reduxjs/toolkit';
+
+interface CalendarLesson {
+  id: number;
+  lessonNumber: number;
+  title: string;
+  duration: number;
+  assessment: boolean;
+  unplugged: boolean;
+  url: string;
+}
+
+export interface CalendarState {
+  showCalendar: boolean;
+  calendarLessons: CalendarLesson[] | null;
+}
+
+const initialState: CalendarState = {
+  showCalendar: false,
+  calendarLessons: null,
+};
+
+const calendarReduxSlice = createSlice({
+  name: 'calendar',
+  initialState,
+  reducers: {
+    setShowCalendar(state, action: PayloadAction<boolean>) {
```

Comment:
Since we are only ever setting both showCalendar and calendarLessons, we could just have one method (e.g. `setCalendarData`). What do you think? This would help with [this](https://redux.js.org/style-guide/#avoid-dispatching-many-actions-sequentially) best practice that I was looking at yesterday